src/public/modules/forms/admin/controllers/edit-fields-modal.client.controller.js

```diff
@@ -551,12 +560,16 @@ function EditFieldsModalController(
       }
     } else if (image) {
       vm.uploadError = null
-
-      FileHandler.uploadImage(
-        image,
-        vm.myform._id,
-        vm.shouldCancelUpload.promise,
-      )
+      source = CancelToken.source()
+
+      return $q
+        .when(
+          uploadImage({
+            image,
+            formId: vm.myform._id,
+            cancelToken: source.token,
+          }),
+        )
```

Comment:
We're able to drop use of `$q` and its outdated syntax, since we polyfill Promises from `core-js`.
```suggestion
      uploadImage({
        image,
        formId: vm.myform._id,
        cancelToken: source.token,
      }),
```

Reply:
$q triggers a digest cycle, whilst using promises by themselves won't.

Reply:
i see, ok. FYI, I see in the docs that `$q.resolve` is an alias of `$q.when` to align with ES6 naming conventions